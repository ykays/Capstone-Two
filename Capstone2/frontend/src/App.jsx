import React, {useState, useEffect} from 'react';
import './App.css'
import NavBar from './NavBar'
import Map from './Map'
import ParksApi from './api'
import useLocalStorageState from './hooks/useLocalStorageState';
import {BrowserRouter, Routes, Route} from "react-router-dom"
import { jwtDecode } from "jwt-decode";
import UserContext from "./UserContext";
import {fetchUserDetailsFromAPI, removeUser} from "./actions/user"
import { useSelector, useDispatch, shallowEqual } from "react-redux";

function App() {
  const dispatch = useDispatch()
  //const userRedux = useSelector(store=>store.user, shallowEqual)

  const [isLoading, setIsLoading] = useState(true);
  const [token, setToken] = useLocalStorageState("token");
  const [user, setUser] = useState();

  async function registerUser(user) {
    try{
      const userToken = await ParksApi.registerUser(user);
      setToken(userToken.token);
      return { success: true };
    }
    catch(err){
      return {success: false, err}
    }
 
  }

  async function logInUser(user) {
    try {
      const userToken = await ParksApi.logInUser(user);
      setToken(userToken.token);
      return { success: true };
    } catch (err) {
      return { success: false, err };
    }
  }

  function logOutUser() {
    dispatch(removeUser());
    setToken(undefined);
    setUser("");
    
  }

  useEffect(() => {
    if (!token || token===undefined) {
      setIsLoading(false);
      return;
    }
    const decoded = jwtDecode(token);
    dispatch(fetchUserDetailsFromAPI(decoded.username))
    setUser((user) => (decoded.username));
  }, [token]);


  return (
    <>
    <UserContext.Provider value={{ user, setUser }}> 
      <NavBar registerUser={registerUser} logInUser={logInUser} logOutUser={logOutUser}/>
      <Map /> 
    </UserContext.Provider>
    
     
     
    </>
  )
}

export default App
