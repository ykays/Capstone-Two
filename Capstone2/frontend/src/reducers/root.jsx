import parks from "./parks";
import filters from "./filters";
import user from "./user";
import routes from "./routes";
import waypoints from "./waypoints";
import { combineReducers } from "redux";

export default combineReducers({
  parks,
  filters,
  user,
  routes,
  waypoints,
});
