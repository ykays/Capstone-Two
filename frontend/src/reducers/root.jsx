import parks from "./parks";
import filters from "./filters";
import user from "./user";
import routes from "./routes";
import { combineReducers } from "redux";

export default combineReducers({
  parks,
  filters,
  user,
  routes,
});
