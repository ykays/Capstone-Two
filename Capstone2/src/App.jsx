import './App.css'

function App() {
  

  return (
    <>
     <h1>Pin the Park App</h1>
    </>
  )
}

export default App
