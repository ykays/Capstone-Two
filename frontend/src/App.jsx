import React, { useState, useEffect } from "react";
import "./App.css";
import NavBar from "./NavBar";
import Map from "./Map";
import ParksApi from "./api";
import useLocalStorageState from "./hooks/useLocalStorageState";
import { jwtDecode } from "jwt-decode";
import { fetchUserDetailsFromAPI, removeUser } from "./actions/user";
import { useDispatch } from "react-redux";

function App() {
  const dispatch = useDispatch();

  const [isLoading, setIsLoading] = useState(true);
  const [token, setToken] = useLocalStorageState("token");

  //function to register user (will be passed to Register Component)
  async function registerUser(user) {
    try {
      const userToken = await ParksApi.registerUser(user);
      setToken(userToken.token);
      return { success: true };
    } catch (err) {
      return { success: false, err };
    }
  }

  //function to register user (will be passed to Login Component)
  async function logInUser(user) {
    try {
      const userToken = await ParksApi.logInUser(user);
      setToken(userToken.token);
      return { success: true };
    } catch (err) {
      console.log(err, "LUAAAERRORAPP");
      return { success: false, err };
    }
  }

  //loging out user (removing from redux store and setting token to undefined in local storage)
  function logOutUser() {
    dispatch(removeUser());
    setToken(undefined);
  }

  // whenever the token changes (register, login, logout) the user details are retrieved and store in redux
  useEffect(() => {
    if (!token || token === undefined) {
      setIsLoading(false);
      return;
    }
    const decoded = jwtDecode(token);
    dispatch(fetchUserDetailsFromAPI(decoded.username));
  }, [token]);

  // setting state for creating new route to be able to manipulate the screen size of entire Map/App
  const [newRoute, setNewRoute] = useState(false);

  return (
    <div style={{ width: newRoute ? "83vw" : "100vw" }}>
      <NavBar
        registerUser={registerUser}
        logInUser={logInUser}
        logOutUser={logOutUser}
      />
      <Map newRoute={newRoute} setNewRoute={setNewRoute} />
    </div>
  );
}

export default App;
