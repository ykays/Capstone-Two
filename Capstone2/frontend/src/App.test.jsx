import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
import { Provider } from "react-redux";
import root from "./reducers/root.jsx";
import { configureStore } from "@reduxjs/toolkit";

const store = configureStore({ reducer: root });
describe("App", () => {
  it("renders the App component without crashing", () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });
  it("matches snapshot", function () {
    const { asFragment } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
