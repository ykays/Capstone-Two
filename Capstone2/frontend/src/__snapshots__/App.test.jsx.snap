// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 100vw;"
  >
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-7e0s21-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PushPinIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 9V4h1c.55 0 1-.45 1-1s-.45-1-1-1H7c-.55 0-1 .45-1 1s.45 1 1 1h1v5c0 1.66-1.34 3-3 3v2h5.97v7l1 1 1-1v-7H19v-2c-1.66 0-3-1.34-3-3"
              fill-rule="evenodd"
            />
          </svg>
           
          <div
            class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
          >
            Pin the Park
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Login
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Register
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
    <div
      class="MuiBox-root css-mdb0go"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
